lib/log_ratelimit.c

```diff
@@ -112,13 +119,8 @@ log_ratelimit_allow(struct log_ratelimit_state *lrs)
 int
 rte_log_ratelimit(uint32_t level, uint32_t logtype, const char *format, ...)
 {
-	struct log_ratelimit_state *lrs;
-	int ratelimit_level = rte_log_get_level(logtype);
-	if (unlikely(ratelimit_level < 0))
-		return -1;
-
-	lrs = &log_ratelimit_states[rte_lcore_id()];
-	if (level <= (typeof(level))ratelimit_level &&
+	struct log_ratelimit_state *lrs = &log_ratelimit_states[rte_lcore_id()];
+	if (level <= (typeof(level))lrs->log_level &&
```

Comment:
Since we no longer need `ratelimit_level`, we can also discard the `typeof(level)`.